integration-test-groups/cxf-soap/cxf-soap-ws-security-server/README.adoc

```diff
@@ -16,4 +16,50 @@ We test in two ways how the SOAP service endpoints are deployed: the Camel way a
 * The service method then forwards to a Camel route defined in `WsSecurityPolicyServerRoutesCxfWay`
 * See also `WssSecurityPolicyHelloServiceCxfWayImpl`
 * This way may come in handy in situations when the Camel way does not work properly,
-  such as https://github.com/apache/camel-quarkus/issues/4291
\ No newline at end of file
+  such as https://github.com/apache/camel-quarkus/issues/4291
+
+= FIPS
+
+Please use profile `fips` if running the tests in the FIPS-enabled environment. The tests have to leverage BouncyCastle-fips dependency instead of standard BouncyCastle.
+
+= Generating keystores
+
+=== Generate keys
+
+```
+openssl genrsa -out alice.key 2048
+openssl genrsa -out bob.key 2048
+```
+
+=== Certificate authority
+
+When prompted for certificate information, confirm default values.
+```
+openssl genrsa -out cxfca.key 2048
+openssl req -x509 -new -key cxfca.key -nodes -out cxfca.pem -config cxfca-openssl.cnf -days 3650 -extensions v3_req
+openssl req -new -subj '/O=apache.org/OU=eng (NOT FOR PRODUCTION)/CN=cxfca' -x509 -key cxfca.key -out cxfca.crt
+```
+
+=== Generate certificates
+```
+openssl req -new -subj '/O=apache.org/OU=eng (NOT FOR PRODUCTION)/CN=cxfca' -x509 -key cxfca.key -out cxfca.crt
+
+openssl req -new -subj '/O=apache.org/OU=eng (NOT FOR PRODUCTION)/CN=alice' -key alice.key -out alice.csr
+openssl x509 -req -in alice.csr -CA cxfca.pem -CAkey cxfca.key -CAcreateserial -out alice.crt
+
+openssl req -new -subj '/O=apache.org/OU=eng (NOT FOR PRODUCTION)/CN=bob' -key bob.key -out bob.csr
+openssl x509 -req -in bob.csr -CA cxfca.pem -CAkey cxfca.key -CAcreateserial -out bob.crt
+```
+
+=== Export keystores
+
+When prompted for password, type `password`.
+When prompted whether to trust the certificate, type `yes`.
+
+```
+openssl pkcs12 -export -in alice.crt -inkey alice.key -certfile cxfca.crt -name "alice" -out alice.p12 -passout pass:password -keypbe aes-256-cbc -certpbe aes-256-cbc
+openssl pkcs12 -export -in bob.crt -inkey bob.key -certfile cxfca.crt -name "bob" -out bob.p12 -passout pass:password -keypbe aes-256-cbc -certpbe aes-256-cbc
+
+keytool -import -trustcacerts -alias bob -file bob.crt -keystore alice.p12
+keytool -import -trustcacerts -alias alice -file alice.crt -keystore bob.p12
```

Comment:
```suggestion
```

This won't work based on the info from https://stackoverflow.com/a/50398011 
It is also not necessary because we do `openssl pkcs12 -export ... -out alice.p12` (and the same for Bob) above.

Reply:
I am blind, sorry, I thought `openssl pkcs12 -export` and `keytool -import` were doing the same thing which they apparently don't. Sorry for the confusion. 

It would be really interesting if `keytool -import` really works. It would mean that keytool is able to import to a proper PKCS12 container, although it is not able to create it.

Reply:
That is my impression, that keytool is able to import to existing PKCS12 container. At least the tests are working and if I reme,mber correctly, this import is necessary - but I'll verify it during today and let you know.

Reply:
Well if it really works, I wonder if we could/should store an empty PKCS12 container created by openssl in the repo and do the rest with keytool plugin. Not that I really insist on doing that. Just an idea. 

If we stay with the shell script, can't the `keytool -import` be done with openssl? The shell script would thus have less dependencies.

Reply:
I understand, I originally tried to achieve that with openssl only,  but without a success . But I've just found a guide describing how to do that. so I  will try, whether the result is  as described,

Reply:
@ppalaga Eventhough it is possible to generate a PKCS12 with more certificates (with a workaround by concatenating the certificates), there is  only 1 alias  present in the keystore. Cxf is not able to load appropriate certificate from such keystore using alias and there is no API on `openssl` CLI for setting different aliases. Therefore the solution with keytool is in my POV the only usable one.

Reply:
OK, thanks for explaining!